### PR TITLE
192 search

### PR DESCRIPTION
**Pull request by chStaiger**
Switch search over to TransferDataThread instead of having an own DownloadThread.

### CODE REVIEW

File: ibridgesgui/gui_utils.py [hunk 1]
```diff
@@ -74,6 +74,26 @@ def prep_session_for_copy(session, error_label) -> pathlib.Path:
     return None
 
 
+def combine_diffs(session, sources: list, destination: Union[IrodsPath, pathlib.Path]) -> dict:
+    """Combine the diffs of several upload or download dry-runs."""
+    diffs = {
```
Comment (qubixes) [inline ibridgesgui/gui_utils.py:79]: I would give this a different name. `diff` and `diffs` are quite similar.

File: ibridgesgui/search.py [hunk 2]
```diff
@@ -101,8 +98,9 @@ def load_results(self, results):
         table_data = []  # (Path, Name, Size, Checksum, created, modified)
         for result in results:
             if "DATA_NAME" in result:
-                obj = IrodsPath(self.session,
-                                result["COLL_NAME"] + "/" + result["DATA_NAME"]).dataobject
+                obj = IrodsPath(
```
Comment (qubixes) [inline ibridgesgui/search.py:101]: Instead of the "/" and "+", you can just use a comma?

File: ibridgesgui/gui_utils.py [hunk 3]
```diff
@@ -74,6 +74,26 @@ def prep_session_for_copy(session, error_label) -> pathlib.Path:
     return None
 
 
+def combine_diffs(session, sources: list, destination: Union[IrodsPath, pathlib.Path]) -> dict:
```
Comment (qubixes) [inline ibridgesgui/gui_utils.py:77]: This function should probably be made obsolete by implementing it in the ibridges library, but for now a useful feature!

**Review by qubixes — CHANGES_REQUESTED** [review]
I haven't clicked the buttons, but reading to code it looks very reasonable.

File: ibridgesgui/gui_utils.py (hunk 3 above)
Comment (chStaiger) [reply ibridgesgui/gui_utils.py:77]: Be my guest ;)

File: ibridgesgui/search.py (hunk 2 above)
Comment (chStaiger) [reply ibridgesgui/search.py:101]: Indeed!

**Pull request merged**